ngraph/frontend/onnx_import/include/onnx_import/editor/editor.hpp

```diff
@@ -0,0 +1,79 @@
+//*****************************************************************************
+// Copyright 2017-2020 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#pragma once
+
+#include <istream>
+#include <map>
+#include <memory>
+
+#include "ngraph/type/element_type.hpp"
+#include "onnx_import/utils/onnx_importer_visibility.hpp"
+
+namespace ONNX_NAMESPACE
+{
+    // forward declaration to avoid the necessity of include paths setting in components
+    // that don't directly depend on the ONNX library
+    class ModelProto;
+} // namespace ONNX_NAMESPACE
+
+namespace ngraph
+{
+    namespace onnx_import
+    {
+        /// \brief A class representing a set of utilities allowing modification of an ONNX model
+        ///
+        /// \note This class can be used to modify an ONNX model before it gets translated to
+        ///       an ngraph::Function by the import_onnx_model function. It lets you modify the
+        ///       model's input types and shapes, extract a subgraph and more. An instance of this
+        ///       class can be passed directly to the onnx_importer API.
+        class ONNX_IMPORTER_API ONNXModelEditor final
```

Comment:
I have a question 😄 
Why do we need this class and who will use it in the Inference Engine?
As I know we use ONNX reader in order to read the ONNX model, will the reader use it? Or not?

Reply:
This class is one of the steps to acchieve MO-parity. It's supposed to be used to adjust the model before it's imported in many ways. This PR only add support for input type changing, but the upcoming work will deliver reshapeability, model cutting and more.